frontend/src/app/component/phenotypic-feature/age-control/date/date.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component } from "@angular/core";
+import { FormControl, FormGroupDirective, NgForm, Validators } from "@angular/forms";
+import { ErrorStateMatcher } from "@angular/material/core";
+
+/** Error when invalid control is dirty, touched, or submitted. */
+export class MyErrorStateMatcher implements ErrorStateMatcher {
+    isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+        const isSubmitted = form && form.submitted;
+        return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+    }
+}
+
+@Component({
+    selector: 'app-date',
+    templateUrl: './date.component.html',
+    styleUrls: ['./date.component.scss']
+})
+
+export class DateComponent {
+
+    yearFormControl = new FormControl('', [
+        Validators.required,
+        Validators.pattern(/^[12][0-9]{3}$/)
```

Comment:
These validators likely need to change as we changed the ranges of year, month, and day fields